ozaria/site/components/teacher-dashboard/modals/ModalDynamicPromotion.vue

```diff
@@ -0,0 +1,98 @@
+<script>
+import Modal from '../../common/Modal'
+import utils from 'core/utils'
+import trackable from 'app/components/mixins/trackable.js'
+
+export default Vue.extend({
+  components: {
+    Modal,
+  },
+  mixins: [trackable],
+  props: {
+    title: {
+      type: String,
+      default: null,
+      required: false
+    },
+    seenPromotionsProperty: {
+      type: String,
+      required: true
+    },
+    cocoOnly: {
+      type: Boolean,
+      default: false
+    },
+    ozarOnly: {
+      type: Boolean,
+      default: false
+    }
+  },
+
+  data: () => {
+    return {
+      showModal: true
+    }
+  },
+
+  computed: {
+    isCodeCombat () {
+      return utils.isCodeCombat
+    },
+    isOzaria () {
+      return utils.isOzaria
+    }
+  },
+
+  watch: {
+    showModal (newVal) {
+      if (this.cocoOnly && this.isOzaria && newVal) {
+        this.hideModal()
+      }
+      if (this.ozarOnly && this.isCodeCombat && newVal) {
+        this.hideModal()
+      }
+    }
+  },
+
+  created () {
+    if (this.seenPromotionsProperty) {
+      this.showModal = !me.getSeenPromotion(this.seenPromotionsProperty)
+    }
+  },
+
+  methods: {
+    close () {
+      if (this.seenPromotionsProperty) {
+        me.setSeenPromotion(this.seenPromotionsProperty)
+      }
+      this.hideModal()
+    },
+    hideModal () {
+      this.showModal = false
+      this.$emit('close')
+    },
+    onClose () {
+      this.close()
+    },
+  }
+})
+</script>
+
+<template>
+  <div>
+    <modal
+      v-if="showModal"
+      :title="title"
+      @close="onClose"
+    >
+      <slot name="content" />
+    </modal>
+    <slot
+      name="opener"
```

Comment:
What will be this used for?

Reply:
It's supposed to work like this, in order to remove all the modal-opening related code from the component that includes the modal.
```
<ModalDynamicPromotion>
  <template #opener><button>CLICK ME TO OPEN MODAL</button></template>
</ModalDynamicPromotion>
```